src/snapshot/SnapshotRegistry.cpp

```diff
@@ -12,6 +12,11 @@ SnapshotRegistry::SnapshotRegistry() {}
 faabric::util::SnapshotData& SnapshotRegistry::getSnapshot(
   const std::string& key)
 {
+    if (key.empty()) {
```

Comment:
AFAICT (I could have missed it), this is not tested in the tests. Could you add a very simple check for attempting to get a snapshot with an empty key?